### PR TITLE
Adding `add_overall()` function

### LINKED ISSUE #3: Add `add_overall()` method for `tbl_adverse_events()` tables.
The method can do grand overall, or even within stratum (when stratum assigned in `tbl_adverse_events()`.

Since we have two by-type options (`tbl_adverse_events(by=, strata=)`, what should the default behavior be?

We could have `add_overall.tbl_adverse_events(by = FALSE, strata= FALSE)` arguments. By default, `add_overall()` re-run the function without the by or strataarguments. But if the user had specified both by and stratain the original call, they could request `add_overall(by = TRUE)` to get an "overall" table that is still split by grade, for example.

### LINKED ISSUE #32: Improve column header when no `by=` variable specified
``` r
library(gtreg)

df_adverse_events %>%
  tbl_adverse_events(
    id = patient_id,
    ae = adverse_event
  ) %>%
  gtsummary::as_kable()
```

| **Adverse Event**             | **TRUE** |
|:------------------------------|:---------|
| Anaemia                       | 7 (70)   |
| Difficult digestion           | 8 (80)   |
| Increased tendency to bruise  | 8 (80)   |
| Intestinal dilatation         | 6 (60)   |
| Iron deficiency anaemia       | 7 (70)   |
| Myochosis                     | 7 (70)   |
| Non-erosive reflux disease    | 6 (60)   |
| Pancreatic enzyme abnormality | 7 (70)   |
| Thrombocytopenia              | 7 (70)   |

<sup>Created on 2021-12-20 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

### PR DESCRIPTION
**Pull request by ddsjoberg**
closes #3
closes #32

- Added the `add_overall()` function.
- The function can add overall stats omitting the by variable, strata variable, or both.
- The function depends on the the dev version of gtsummary
    - We cannot submit to CRAN while we depend on the dev version
    - I added a `Remotes:` field in DESCRIPTION indicating the install of gtsummary from GitHub
- To make this function work, I needed to update `tbl_adverse_event()` and assign unique variable names to each piece of the stacked and merge tbl. Without these updates, there was an issue trying to merge non-unique IDs.
- I also updated the default header when no by variable is present to "Overall"

Let me know what you think of the API of `add_overall()`. Can it be made better? Does it need more functionality?

### CODE REVIEW

**Review by shannonpileggi — CHANGES_REQUESTED** [review]
Overall (hehe) this looks amazing!!

I wonder if users will want a couple of more additional overall nesting options that are I don't think are yet available (pls correct me if I am wrong).

This might include:
1. A single overall column for across all `strata` and `by` when `strata = TRUE` is specified.
image


2. Additional overall columns across all strata and by when `by = TRUE` is specified.
image

`add_overall()` does currently suit my needs 🎉 so I am also happy to leave this as an open issue for future development if other users desire this feature.

3. In line with issue #41, we would want tables like this to have an `N` displayed for overall as well. The way the headings line up I wonder if that would be problematic as `Overall` is not at the same position as `strata`?

image

What an amazing feat you have already accomplished so quickly!! 🙌

File: R/reexport.R [hunk 1]
```diff
@@ -86,3 +86,7 @@ gtsummary::as_kable_extra
 #' @export
 gtsummary::inline_text
 
+#' @importFrom gtsummary add_overall
+#' @export
+gtsummary::add_overall
```
[the remark this answers is not in the document]
Comment (ddsjoberg) [reply R/reexport.R:91]: YES! Let's make a list of items to discuss when we're back to work :)

File: R/complete_ae_data.R [hunk 2]
```diff
@@ -100,7 +100,7 @@
 
   # configuring the `by=` variable ---------------------------------------------
   if (is.null(by)) {
-    data$by <- factor("TRUE")
+    data$by <- factor("Overall")
```
[the remark this answers is not in the document]
Comment (ddsjoberg) [reply R/complete_ae_data.R:103]: The users can update this now with `modify_header()`. Couple of points about using this function:
1. In gtsummary, the tables are mostly simple with simple column names. But after all the stratifying and merging, the column names have become harder to predict. That would indicate that adding the option as a function argument would be preferable. 
2. BUT, I LOVE to keep functions simple with few arguments. When I see a new function/package, and there are too many args, I get overwhelmed.

For now, let's require users to use `modify_header()`, and when we get closer to the end of the development we can re-visit?

**Comment by ddsjoberg** [comment]
You raise good points! Being able to run `add_overall()` multiple times would be helpful.

I did a bit of a lazy implementation, using `tbl_merge()` at the end. While this creates a nice looking table, it changes the underlying structure of the table. I should do something to make it more flexible....it's just hard because the users have various options for by-type variables in the original table (by= and strata=) and the same in add_overall(). Depending on their choices, the columns names can be different.

I'll think on it and submit some updates!

**Comment by ddsjoberg** [comment]
> In line with issue #41, we would want tables like this to have an N displayed for overall as well. The way the headings line up I wonder if that would be problematic as Overall is not at the same position as strata?

I was thinking about how best to add the Ns, and I was getting confused. Ns in the strata row make perfect sense, and it's clear what should go there.

But if I have a table `tbl_adverse_event(by=) %>% add_overall()`, both sets of tables are based off the same number of patients. So why would we only put the N on the overall column?

I suggest we go ahead merge this for now, and I'll update `add_overall()` to be able to run multiple times in a separate PR. We can also think more how the Ns will/will not be placed.

File: R/complete_ae_data.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (shannonpileggi) [reply R/complete_ae_data.R:103]: Sounds great!

**Pull request merged**